File: Python/rotate_matrix_90.py
def rotate_90(matrix):
    n = len(matrix)
    for i in range(n):
        for j in range(i, n):
            matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
    for row in matrix:
        row.reverse()
    return matrix

# Input from user
n = int(input("Enter size of square matrix (n x n): "))
matrix = []

print("Enter matrix row by row:")
for i in range(n):
    row = list(map(int, input().split()))
    matrix.append(row)

rotated = rotate_90(matrix)
print("Matrix after 90° rotation:")
for row in rotated:
    print(*row)

File: C/rotate_matrix_90.c
#include <stdio.h>

void rotateMatrix(int rows, int cols, int matrix[rows][cols]) {
    // First transpose the matrix
    for(int i = 0; i < rows; i++) {
        for(int j = i; j < cols; j++) {
            int temp = matrix[i][j];
            matrix[i][j] = matrix[j][i];
            matrix[j][i] = temp;
        }
    }
    
    // Then reverse each row
    for(int i = 0; i < rows; i++) {
        for(int j = 0; j < cols/2; j++) {
            int temp = matrix[i][j];
            matrix[i][j] = matrix[i][cols-1-j];
            matrix[i][cols-1-j] = temp;
        }
    }
}

int main() {
    int rows, cols;
    printf("Enter matrix dimensions: ");
    scanf("%d %d", &rows, &cols);
    
    int matrix[rows][cols];
    printf("Enter matrix elements:\n");
    for(int i = 0; i < rows; i++)
        for(int j = 0; j < cols; j++)
            scanf("%d", &matrix[i][j]);
    
    printf("\nOriginal Matrix:\n");
    for(int i = 0; i < rows; i++) {
        for(int j = 0; j < cols; j++)
            printf("%d ", matrix[i][j]);
        printf("\n");
    }
            
    rotateMatrix(rows, cols, matrix);
    
    printf("\nRotated Matrix:\n");
    for(int i = 0; i < rows; i++) {
        for(int j = 0; j < cols; j++)
            printf("%d ", matrix[i][j]);
        printf("\n");
    }
    return 0;
}